### PR TITLE
Fcs 149

### PR DESCRIPTION
**Pull request by JamilOmar**
- [x] Support for cliPattern instead of cliDir

- [x] Support for dependencies as dictionary

- [x] Support for loading config folder and config.json files

### CODE REVIEW

File: test/lib/unit/cli/init_spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  it('tests the init command with a configuration folder and test settings', async () => {
+    const cwd = path.join(process.cwd(), 'test', 'configuration-folder');
+    const config: any = initConfig({
+      cwd,
+      configuration: 'test',
+    });
+    expect(config).toBeDefined();
+    expect(config.test).toEqual('test from folder with test config');
+  });
+});
```
Comment (dkrantsberg) [inline test/lib/unit/cli/init_spec.ts]: Can you add a test to check that environment variables are being set as expected.

File: lib/cli/init.ts [hunk 2]
```diff
@@ -54,3 +62,42 @@ export function init(): void {
...
+    });
+  } else if (
+    fs.existsSync(configFolderPath) &&
+    fs.lstatSync(configFolderPath).isDirectory()
+  ) {
+    if (!_.isNil(opts.configuration)) {
+      process.env.NODE_CONFIG_ENV = opts.configuration;
+    }
+    if (fs.existsSync(envFilePath)) {
+      require('dotenv').config({
```
Comment (dkrantsberg) [inline lib/cli/init.ts]: The coverage still says that this line is not getting hit. 
https://codecov.io/gh/LabShare/lsc/pull/163/diff?src=pr&el=tree#D10-95

Does it actually load set environment variables from .env file in tests?

Comment (JamilOmar) [reply lib/cli/init.ts]: ahh will add that test as well

**Pull request merged**